src/map.cpp

```diff
@@ -69,32 +64,18 @@ extern "C" _CRTIMP void __cdecl _assert(void *, void *, unsigned);
 
 
 #ifdef _DEBUG
-TileIndex TileAdd(TileIndex tile, TileIndexDiff add,
-	const char *exp, const char *file, int line)
+TileIndex TILE_ADD(TileIndex tile, TileIndexDiff offset, std::source_location location)
 {
-	int dx;
-	int dy;
-	uint x;
-	uint y;
-
-	dx = add & Map::MaxX();
+	int dx = offset & Map::MaxX();
```

Comment:
While at it, can't we use `TileX(offset)` and `TileY(offset)` here?

I was mostly triggered by `dy`, which looks a bit more complicated than it has to. `TileY` does it less complicated :D

(not your change ofc, but as we are here anyway)

Reply:
Though `offset` isn't of type `TileIndex`, so technically `TileX` and friends are not the right function to call. Also `TileX` returns an unsigned int, whereas this needs a signed int.

Reply:
Owh, right, that weird `TileIndexDiff` that is only used in, what, 50 instances? I forgot about that :P We also have `TileIndexDiffC`, solving the same problem, but differently. I forgot about those weird types. Nothing for this PR.